runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1442,15 +1442,14 @@ TR_J9ServerVM::isClassVisible(TR_OpaqueClassBlock *sourceClass, TR_OpaqueClassBl
    return std::get<0>(stream->read<bool>());
    }
 
+// Check a map of registered thunks for this client.
+// If a pointer to client-side thunk is there, return it.
+// If a thunk is not registered, send a message to the client to check if it's registered there,
+// cache the result if it is. Note that it's possible to have a thunk registered on the client
+// but not in this map, when client reconnects to a different server.
 void *
-TR_J9ServerVM::getJ2IThunk(char *signatureChars, uint32_t signatureLength, TR::Compilation *comp)
+TR_J9ServerVM::getClientJ2IThunk(std::string signature, TR::Compilation *comp)
```

Comment:
The signature should be passed as either pointer and size or by `const` reference, definitely not as `std::string` by value.